src/DynamoCoreWpf/Views/CodeBlocks/CodeBlockEditor.cs

```diff
@@ -7,12 +7,56 @@
 namespace Dynamo.UI.Controls
 {
     /// <summary>
+    /// Code block node editor state machine.
+    /// </summary>
+    internal class EditorStateMachine
+    {
+        internal enum EditorState
+        {
+            Editing,
+            Commited
+        }
+
+        private EditorState state;
+
+        /// <summary>
+        /// Event handler for editor state changed.
+        /// </summary>
+        public event Action<EditorState> OnStateChanged;
```

Comment:
Not sure if this and the following `public` ones make it into documentation since the class itself is `internal`. I'm wondering if we can make them `internal` too?